src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -22,6 +25,42 @@ IShaderCompiler::IShaderCompiler(core::smart_refctd_ptr<system::ISystem>&& syste
     m_defaultIncludeFinder = core::make_smart_refctd_ptr<CIncludeFinder>(core::smart_refctd_ptr(m_system));
 }
 
+inline core::smart_refctd_ptr<ICPUShader> nbl::asset::IShaderCompiler::compileToSPIRV(const std::string_view code, const SCompilerOptions& options) const
+{
+    CCache::SEntry entry;
+    std::vector<CCache::SEntry::SPreprocessingDependency> dependencies;
+    if (options.readCache || options.writeCache)
+        entry = std::move(CCache::SEntry(code, options));
```

Comment:
you don't need the std::move here, `CCache::SEntry(code, options)` is already an r-value